### PR TITLE
[invalid] Test x64-linux-dynamic

### PR DESCRIPTION
**Pull request by Osyotr**
Towards #25668

### CODE REVIEW

**Comment by BillyONeal** [comment]
This looks correct to me so I'm going to cherry pick it into https://github.com/microsoft/vcpkg/pull/37129 to coalesce world rebuilds

**Comment by Osyotr** [comment]
@BillyONeal don't forget about `ports.cmake` otherwise this will change nothing

File: scripts/cmake/z_vcpkg_fixup_rpath.cmake [hunk 1]
```diff
@@ -116,8 +116,12 @@ function(z_vcpkg_fixup_rpath_in_dir)
                 ERROR_VARIABLE set_rpath_error
             )
 
-            message(STATUS "Adjusted RPATH of '${elf_file}' (From '${org_rpath}' -> To '${new_rpath}')")
+            if(NOT "${set_rpath_error}" STREQUAL "")
+                message(WARNING "Couldn't adjust RPATH of '${elf_file}': ${set_rpath_error}")
+                continue()
+            endif()
 
+            message(STATUS "Adjusted RPATH of '${elf_file}' (From '${org_rpath}' -> To '${new_rpath}')")
```
Comment (Neumann-A) [inline scripts/cmake/z_vcpkg_fixup_rpath.cmake:124]:
```suggestion
            message(STATUS "Adjusted RPATH of '${elf_file}' (From '${readelf_output}' -> To '${new_rpath}')")
```

File: ports/curlpp/portfile.cmake [hunk 2]
```diff
@@ -26,7 +26,7 @@ if(VCPKG_LIBRARY_LINKAGE STREQUAL static)
     )
 endif()
 
-if(VCPKG_TARGET_IS_WINDOWS AND NOT (VCPKG_LIBRARY_LINKAGE STREQUAL static))
+if(NOT (VCPKG_LIBRARY_LINKAGE STREQUAL static))
```
Comment (dg0yt) [inline ports/curlpp/portfile.cmake:29]: Do we need a condition at all?

File: ports/alsa/portfile.cmake [hunk 3]
```diff
@@ -19,6 +19,7 @@ vcpkg_from_github(
     HEAD_REF master
     PATCHES
         "fix-plugin-dir.patch"
+        "fix-linux-dynamic.patch" # https://github.com/alsa-project/alsa-lib/commit/2a736a0d2543f206fd2653aaae8a08a4c42eb917
```
Comment (dg0yt) [inline ports/alsa/portfile.cmake:22]: With this comment, I wouldn't need the first 19 lines of the patch.

File: ports/yasm/portfile.cmake [hunk 4]
```diff
@@ -48,13 +48,13 @@ if (BUILD_TOOLS)
...
     vcpkg_copy_tools(TOOL_NAMES vsyasm yasm ytasm ${EXTRA_BUILD_TOOLS} AUTO_CLEAN)
-    if (VCPKG_LIBRARY_LINKAGE STREQUAL "dynamic")
+    if (VCPKG_TARGET_IS_WINDOWS AND VCPKG_LIBRARY_LINKAGE STREQUAL "dynamic")
         file(COPY "${CURRENT_PACKAGES_DIR}/bin/yasmstd${VCPKG_TARGET_SHARED_LIBRARY_SUFFIX}"
             DESTINATION "${CURRENT_PACKAGES_DIR}/tools/${PORT}")
     endif()
 endif()
 
-if(VCPKG_LIBRARY_LINKAGE STREQUAL static)
+if(VCPKG_TARGET_IS_WINDOWS AND VCPKG_LIBRARY_LINKAGE STREQUAL static)
```
Comment (dg0yt) [inline ports/yasm/portfile.cmake:57]:
```suggestion
if(VCPKG_TARGET_IS_WINDOWS AND VCPKG_LIBRARY_LINKAGE STREQUAL "static")
```

**Review by dg0yt — COMMENTED** [review]
(nit-pick)

**Pull request closed without merging**